utils/tangleid/BaseExtension.js

```diff
@@ -0,0 +1,31 @@
+
+/**
+ * Class representing a base extension.
+ */
+class BaseExtension {
+  /**
+   * Create a extension instance.
+   * @param {string} namespace - The namespace of the extension.
+   */
```

Comment:
In this PR, I'm trying to use the [JSDoc](http://usejsdoc.org/howto-es2015-classes.html) to provide code documentation. 

It would be great if we have good code documentation.

Reply:
Mention it in style guide as well.